Check for ActionMailer::Base.mail before proceeding

justincampbell opened the pull request.

justincampbell committed:
Check for ActionMailer::Base.mail before proceeding

bcobb commented on lib/capistrano/notifier/mail.rb:
```diff
@@ -18,7 +18,21 @@ def self.load_into(configuration)
 
   def perform
-    mail = ActionMailer::Base.mail({
+    if Object.const_defined?('ActionMailer') && ActionMailer.const_defined?('Base') && ActionMailer::Base.respond_to?(:mail)
```
You could also do `defined? ActionMailer::Base`

justincampbell replied:
That sounds much better ;) I made the assumption that having the constant in the code would invoke it and fail.

bcobb committed:
Require ActionMailer to allow for Rails 2.

bcobb committed:
Test the sender and recipients of the mail notification

- I also moved the debugging output to the cap task itself, as the only
  time said behavior is desirable is when running the task.
